eng/testing/linker/trimmingTests.targets

```diff
@@ -92,15 +92,13 @@
 
   <Target Name="PublishTrimmedProjects"
           DependsOnTargets="GenerateProjects">
-    <PropertyGroup>
-      <TestPublishTrimmedCommand>"$(DotNetTool)"</TestPublishTrimmedCommand>
-      <TestPublishTrimmedCommand>$(TestPublishTrimmedCommand) build /t:PublishTrimmed</TestPublishTrimmedCommand>
-      <TestPublishTrimmedCommand>$(TestPublishTrimmedCommand) /nr:false</TestPublishTrimmedCommand>
-      <TestPublishTrimmedCommand>$(TestPublishTrimmedCommand) /warnaserror</TestPublishTrimmedCommand>
-      <TestPublishTrimmedCommand>$(TestPublishTrimmedCommand) -p:configuration=$(Configuration)</TestPublishTrimmedCommand>
-    </PropertyGroup>
+    <MSBuild Projects="@(TestConsoleApps)"
+             Targets="Restore"
+             Properties="MSBuildRestoreSessionId=$([System.Guid]::NewGuid());Configuration=$(Configuration);@(CliEnvironment)" />
 
-    <Exec Command="$(TestPublishTrimmedCommand) %(TestConsoleApps.AdditionalArgs)" EnvironmentVariables="@(CliEnvironment)" StandardOutputImportance="Low" WorkingDirectory="%(TestConsoleApps.ProjectDir)" />
+    <MSBuild Projects="@(TestConsoleApps)"
+             Targets="Publish"
```

Comment:
I assume so, but by movint this in-proc do we still get to pick the version of the linker that is used to trim the apps? What happens if the parent msbuild already has the linker task loaded into it (since it links out the shared framework) and the versions of the linker we use for runtime and the framework are different?

Reply:
Good point. To guarantee that, the `ILLinkTasksAssembly` property which determines the assembly to use for the ILLink task, would either need to be passed in globally or as early as possible (i.e. either in Arcade's Build.proj or in dotnet/runtime's Build.proj). @dsplaisted is there a better way to control which ILLink version the sdk uses? I noticed that the sdk imports the ILLink sdk which makes me wonder if it is retrieved from the sdk directory or via nuget via an implicit global.json sdk value? https://github.com/dotnet/sdk/blob/4191572924db60039c2b746f7d261070a056932e/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L14

Reply:
The ILLink MSBuild SDK (Microsoft.NET.ILLink.Tasks) is part of the .NET SDK, it's only a separate MSBuild SDK for codeflow / engineering reasons.  So it should use the ILLink that shipped with the .NET SDK unless you override it.

Reply:
I assumed that such `<Import Project="x.props" Sdk="x" />` use the NuGetSdkResolver. Does a fallback exist that looks into the sdk directory first before probing the nuget cache? cc @nkolev92

Reply:
I submitted https://github.com/dotnet/arcade/pull/6936 to arcade to make sure that the right linker is used (which is very similar to the compiler package opt-in).

Reply:
@joperezr @eerhardt the fix to guarantee that a newer ILLink package is used over the one that ships with the SDK is in dotnet/arcade#6936 but that one requires a newer SDK build because ILLink props and targets aren't sequenced correctly, which will be fixed by https://github.com/dotnet/sdk/pull/15801. I don't know yet if we can still get that change into preview 1.

Reply:
This generally won't use the NuGet Sdk Resolver: `<Import Project="x.props" Sdk="x" />`

For the NuGet resolver to be used, the version for the SDK to use has to be specified.  It might be possible to do that in a global.json folder, but I think the .NET SDK resolver would still take precedence.

Reply:
Got it. I also found https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-use-project-sdk?view=vs-2019#how-project-sdks-are-resolved which goes into a bit more detail.

I just tried specify a version for ILLink in an app's global.json but it isn't honored presumably because the .NET SDK resolver takes precedence (as you pointed out above).